app/assets/javascripts/Patient.js

```diff
@@ -0,0 +1,143 @@
+var Schema = mongoose.Schema;
+var Code = Schema.Types.Code;
+var Interval = Schema.Types.Interval;
+var Quantity = Schema.Types.Quantity;
+
+var PatientSchema = new Schema({
+  birth_datetime: DateTime,
+  qdm_version: { type: String, default: "5.3" },
+  ethnicity: Code,
+  race: Code,
+  sex: Code,
+  // These are the "data criteria", or QDM datatype elements that exist on a
+  // patient.
+  history_elements: []
```

Comment:
It looks like everything is being stored in an array. Will this cause access efficiency issues over time? Does it make sense to store these in a hash based on the data type or oid of the data type?

Reply:
I need to double check, but I believe this is just how mongoose does `embeds_many`.